File: main/src/main/java/net/fwitz/math/main/binary/dual/functions/examples/simple/DualExpLogZ.java
package net.fwitz.math.main.binary.dual.functions.examples.simple;

import net.fwitz.math.plot.binary.dual.DualNumberFunctionPlot;

public class DualExpLogZ {
    public static void main(String[] args) {
        new DualNumberFunctionPlot("e^(log z)")
                .computeFn(w -> w.region1mapped(z -> z.log().exp()))
                .render();
    }
}

File: main/src/main/kotlin/net/fwitz/math/main/binary/dual/functions/examples/simple/DualExpLogZ.kt
package net.fwitz.math.main.binary.dual.functions.examples.simple

import net.fwitz.math.binary.dual.DualNumber
import net.fwitz.math.plot.binary.dual.DualNumberFunctionPlot

object DualExpLogZ {
    @JvmStatic
    fun main(args: Array<String>) = DualNumberFunctionPlot("e^(log z)")
        .computeFn { w: DualNumber -> w.region1mapped { z -> z.log.exp } }
        .render()
}